src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -22,6 +22,87 @@ private enum EscapeLevel : uint
             ReturnOnly = ReturnOnlyScope,
         }
 
+        /// <summary>
+        /// Encapsulates a property or indexer and the accessor(s) on it that were used. The particular
+        /// indexer used is important as it impacts ref safety analysis
+        /// </summary>
+        private readonly struct MethodInfo
+        {
+            internal Symbol Symbol { get; }
+            internal MethodSymbol? Method { get; }
+            internal MethodSymbol? SetMethod { get; }
+            internal bool IsCompoundUsage => SetMethod is not null;
+            internal bool UseUpdatedEscapeRules => Method?.UseUpdatedEscapeRules == true;
+            internal bool ReturnsRefToRefStruct =>
+                Method is { RefKind: not RefKind.None, ReturnType: { } returnType } &&
+               returnType.IsRefLikeOrAllowsRefLikeType();
+
+            private MethodInfo(Symbol symbol, MethodSymbol? method, MethodSymbol? setMethod)
+            {
+                Symbol = symbol;
+                Method = method;
+                SetMethod = setMethod;
+            }
+
+            internal static MethodInfo Create(MethodSymbol method)
+            {
+                return new MethodInfo(method, method, null);
+            }
+
+            internal static MethodInfo Create(PropertySymbol property)
+            {
+                return new MethodInfo(
+                    property,
+                    property.GetMethod ?? property.SetMethod,
```

Comment:
Why do we use `GetMethod` and `SetMethod` here but `GetOwnOrInheritedGetMethod()` and `GetOwnOrInheritedSetMethod()` below? Consider documenting if the difference is intentional.

Reply:
It's not intentional. Will normalize.